components/chat_widget/src/components/ocs-chat/ocs-chat.tsx

```diff
@@ -606,6 +623,92 @@ export class OcsChat {
     this.initializePosition();
   };
 
+  private getStorageKeys() {
+    return {
+      sessionId: `ocs-chat-session-${this.chatbotId}`,
+      messages: `ocs-chat-messages-${this.chatbotId}`,
+      lastActivity: `ocs-chat-activity-${this.chatbotId}`
+    };
+  }
+
+  private saveSessionToStorage(): void {
+    const keys = this.getStorageKeys();
+    try {
+      if (this.sessionId) {
+        localStorage.setItem(keys.sessionId, this.sessionId);
+        localStorage.setItem(keys.lastActivity, new Date().toISOString());
+      }
+      localStorage.setItem(keys.messages, JSON.stringify(this.messages));
+    } catch (error) {
+      console.warn('Failed to save chat session to localStorage:', error);
+    }
+  }
+
+  private loadSessionFromStorage(): { sessionId?: string; messages: ChatMessage[] } {
+    const keys = this.getStorageKeys();
+    try {
+      const sessionId = localStorage.getItem(keys.sessionId) || undefined;
+      const messagesJson = localStorage.getItem(keys.messages);
+      const messages = messagesJson ? JSON.parse(messagesJson) : [];
+
+      const lastActivity = localStorage.getItem(keys.lastActivity);
+      if (lastActivity) {
+        const lastActivityDate = new Date(lastActivity);
+        const hoursSinceActivity = (Date.now() - lastActivityDate.getTime()) / (1000 * 60 * 60);
+        if (hoursSinceActivity > 24) {
+          this.clearSessionStorage();
+          return { messages: [] };
+        }
+      }
+
+      return { sessionId, messages };
+    } catch (error) {
+      // fall back to starting a new session
+      console.warn('Failed to load chat session from localStorage, starting new session:', error);
+      return { messages: [] };
+    }
+  }
+
+  private clearSessionStorage(): void {
+    const keys = this.getStorageKeys();
+    try {
+      localStorage.removeItem(keys.sessionId);
+      localStorage.removeItem(keys.messages);
+      localStorage.removeItem(keys.lastActivity);
+    } catch (error) {
+      console.warn('Failed to clear chat session from localStorage:', error);
+    }
+  }
+
+  private isLocalStorageAvailable(): boolean {
+    try {
+      const testKey = '__ocs_test__';
+      localStorage.setItem(testKey, 'test');
+      localStorage.removeItem(testKey);
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  private async startNewChat(): Promise<void> {
+    try {
+      // Clear current session from storage (if available)
+      this.clearSessionStorage();
+    } catch (error) {
```

Comment:
Looks like `clearSessionStorage` already does some error handling, so no need to do it here as well I suppose